mathjax3-ts/components/latest.ts

```diff
@@ -0,0 +1,398 @@
+    }],
+
+    ['cdn.jsdelivr.net', {
+        api: 'https://api.github.com/repos/mathjax/mathjax/releases/latest',
+        key: 'tag_name',
+        base: 'https://cdn.jsdelivr.net/npm/mathjax@'
+    }]
+]);
+
+/**
+ * The data for getting release versions from GitHub
+ */
+const GITHUB: CdnData = {
+    api: 'https://api.github.com/repos/mathjax/mathjax/releases',
+    key: 'tag_name'
+};
+
+/**
+ * The major version number for MathJax (we will load the highest version with this initial number)
+ */
+const MJX_VERSION = 3;
+
+/**
+ * The name to use for the version in localStorage
+ */
+const MJX_LATEST = 'mjx-latest-version';
+
+/**
+ * The amount of time a cached version number is valid
+ */
+const SAVE_TIME = 1000 * 60 * 60 * 24 * 7;   // one week
+
+/**
+ *  data for the script that loaded latest.js
+ */
+let script: ScriptData = null;
+
+/*=====================================================================*/
+
+/**
+ * Produce an error message on the console
+ *
+ * @param {string} message   The error message to display
+ */
+function Error(message: string) {
+    if (console && console.error) console.error('MathJax(latest.js): ' + message);
+}
+
+/**
+ * Create a ScriptData object from the given script tag and CDN
+ *
+ * @param {HTMLScriptElement} script   The script tag whose data is desired
+ * @param {CdnData} cdn                The CDN data already obtained for the script (or null)
+ * @return {ScriptData}                The data for the given script
+ */
+function scriptData(script: HTMLScriptElement, cdn: CdnData = null) {
+    script.parentNode.removeChild(script);
+    let src = script.src;
+    let file = src.replace(/.*?\/latest\.js(\?|$)/, '');
+    if (file === '') {
+      file = 'startup.js';
+      src = src.replace(/\?$/, '') + '?' + file;
+    }
+    const version = (src.match(/(\d+\.\d+\.\d+)\/latest.js\?/) || ['', ''])[1];
+    return {
+        tag: script,
+        src: src,
+        id: script.id,
+        version: version,
+        file: file,
+        cdn: cdn
+    } as ScriptData;
+}
+
+/**
+ * Check if a script refers to MathJax on one of the CDNs
+ *
+ * @param {HTMLScriptElement} script   The script tag to check
+ * @return {boolean}                   True if the script is from a MathJax CDN
+ */
+function checkScript(script: HTMLScriptElement) {
+    for (const server of CDN.keys()) {
+        const cdn = CDN.get(server);
+        const url = cdn.base;
+        const src = script.src;
+        if (src && src.substr(0, url.length) === url && src.match(/\/latest\.js(\?|$)/)) {
+            return scriptData(script, cdn);
+        }
+    }
+    return null;
+}
+
+/**
+ * @return {ScriptData}   The data for the script tag that loaded latest.js
+ */
+function getScript() {
+    if (document.currentScript) {
+        return scriptData(document.currentScript as HTMLScriptElement);
+    }
+    const script = document.getElementById('MathJax-script') as HTMLScriptElement;
```

Comment:
Does that mean `MathJax-script` can be given in any tag, even non-`script` ones?

Reply:
I will add a check that it is on a `script` tag.  Note that this is only for compatibility with old browsers (namely IE), since modern browsers would have already found the script via `currentScript` above).